python/recipe.cpp

```diff
@@ -58,9 +91,12 @@ std::vector<arb::event_generator> convert_gen(std::vector<pybind11::object> pyge
 }
 
 std::vector<arb::event_generator> py_recipe_shim::event_generators(arb::cell_gid_type gid) const {
-    return try_catch_pyexception(
-                [&](){ return convert_gen(impl_->event_generators(gid), gid); },
-                "Python error already thrown");
+    return try_catch_pyexception([&](){
+        pybind11::gil_scoped_acquire guard;
+        auto result = convert_gen(impl_->event_generators(gid), gid);
```

Comment:
Any reason why we don't just `return convert_gen(...);`?

Reply:
Good point.